src/foam/nanos/crunch/ruler/SetUCJStatusOnPut.js

```diff
@@ -41,7 +42,8 @@ foam.CLASS({
             // the following should be checked if the result of previous rule ( validateUCJDataOnPut ) 
             // is not ACTION_REQUIRED. In the ACTION_REQUIRED case, the ucj should be put into the
             // dao without any additional checks
-            Capability capability = (Capability) ucj.findTargetId(x);
+            DAO capabilityDAO = (DAO) x.get("capabilityDAO");
+            Capability capability = (Capability) ucj.findTargetId(((ProxyDAO) capabilityDAO).getX());
```

Comment:
Could you please test and let me know if this would also work?
  Capability capability = (Capability) ucj.findTargetId(ruler.getX());

Thanks